lib/hooks/useTime.ts

```diff
@@ -22,13 +26,15 @@ const useTime = (latitude?: number, longitude?: number) => {
   const sunsetTime = times ? times.sunset : null;
 
   useEffect(() => {
-    const intervalId = setInterval(() => {
-      setCurrentTime(Date.now());
-    }, 1000);
-
-    // clear interval on re-render to avoid memory leaks
-    return () => clearInterval(intervalId);
-  });
+    if (enabled) {
+      const intervalId = setInterval(() => {
+        setCurrentTime(Date.now());
+      }, 1000);
+
+      // clear interval on re-render to avoid memory leaks
+      return () => clearInterval(intervalId);
+    }
+  }, [enabled]);
```

Comment:
Another thing to note is the implementation of `enabled` for `useTime`. `useTime`'s interval can now be disabled to prevent rerendering.

Reply:
could we use a more descriptive name? perhaps something like `updateEnabled` might suffice.